### PR TITLE
Support converting RowBatch and RowBlockV2 to/from Arrow

### PR DESCRIPTION
**Pull request by imay**
For #1703

### CODE REVIEW

File: be/src/util/arrow/row_block.cpp [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+Status convert_to_arrow_batch(const RowBlockV2& block,
+                              const std::shared_ptr<arrow::Schema>& schema,
+                              arrow::MemoryPool* pool,
+                              std::shared_ptr<arrow::RecordBatch>* result) {
+    FromRowBlockConverter converter(block, schema, pool);
+    return converter.convert(result);
+}
+
+// Converte Arrow RecordBatch to Doris RowBlockV2
```
Comment (morningman) [inline be/src/util/arrow/row_block.cpp]:
```suggestion
// Convert Arrow RecordBatch to Doris RowBlockV2
```

File: be/src/util/arrow/row_batch.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+namespace doris {
+
+class MemTracker;
+class ObjectPool;
+class RowBatch;
+class RowDescriptor;
+
+// Convert Doris RowDescriptor to Arrow Schema.
```
Comment (morningman) [inline be/src/util/arrow/row_batch.h]: Better to add comment to tell the difference between row_batch.h and row_block.h

File: be/src/util/arrow/utils.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#include "common/status.h"
+
+// This files contains some utilities to convert Doris internal
+// data format into/from Apache Arrow format. When Doris needs
+// to share data with others we can leverage Arrow's ability.
+// We can convert our internal data into Arrow's memory format
+// and then convert to other format, for example Parquet. Because
+// Arrow have already implemented this functions. And what's more
+// Arrow support multiple laguages, so it will make it easy to
+// handle Doris data in other laguages.
```
[the remark this answers is not in the document]
Comment (imay) [reply be/src/util/arrow/utils.h]: OK

File: be/src/util/arrow/row_block.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+Status convert_to_arrow_schema(
+    const Schema& row_desc,
+    std::shared_ptr<arrow::Schema>* result);
+
+// Convert Arrow Schema to Doris Schema.
+Status convert_to_doris_schema(
+    const arrow::Schema& schema,
+    std::shared_ptr<Schema>* result);
+
+// Converte a Doris RowBlockV2 to an Arrow RecordBatch. A valid Arrow Schema
```
[the remark this answers is not in the document]
Comment (imay) [reply be/src/util/arrow/row_block.h]: OK

File: be/src/util/arrow/row_batch.h (hunk 2 above)
Comment (imay) [reply be/src/util/arrow/row_batch.h]: OK

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**